lib/web_ui/lib/src/engine/text_editing.dart

```diff
@@ -436,93 +423,23 @@ class TextEditingElement {
     domElement.focus();
   }
 
-  /// Swap out the current DOM element and replace it with a new one of type
-  /// [newElementType].
-  ///
-  /// Ideally, swapping the underlying DOM element should be seamless to the
-  /// user of this class.
-  ///
-  /// See also:
-  ///
-  /// * [PersistentTextEditingElement._swapDomElement], which notifies its users
-  ///   that the element has been swapped.
-  void _swapDomElement(ElementType newElementType) {
-    // TODO(mdebbar): Create the appropriate dom element and initialize it.
-  }
-
   void _handleChange(html.Event event) {
-    _lastEditingState = calculateEditingState();
-    _onChange(_lastEditingState);
+    EditingState newEditingState = calculateEditingState();
+    if (newEditingState != _lastEditingState) {
```

Comment:
`EditingState` does not implement `operator ==`, so this only compares identity of two objects. Is that what we want to do here?

Reply:
It does implement `operator ==`.

Reply:
Ah, nvm then. I got confused by Github UI and didn't find it.

Reply:
ack.